plugins/tracker-resources/src/components/issues/Move.svelte

```diff
@@ -0,0 +1,172 @@
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { getClient } from '@hcengineering/presentation'
+  import { Button, Label, Status as StatusControl } from '@hcengineering/ui'
+
+  import core, { Class, Client, Doc, Ref, SortingOrder } from '@hcengineering/core'
+  import { getResource, OK, Resource, Status, translate } from '@hcengineering/platform'
+  import { SpaceSelect } from '@hcengineering/presentation'
+  import { calcRank } from '@hcengineering/task'
+  import { createEventDispatcher } from 'svelte'
+  import view from '@hcengineering/view'
+  import tracker from '../../plugin'
+  import { Issue, Project } from '@hcengineering/tracker'
+  import { moveIssueToSpace } from '../../utils'
+
+  export let selected: Issue | Issue[]
+  $: docs = Array.isArray(selected) ? selected : [selected]
+
+  let status: Status = OK
+  let currentSpace: Project | undefined
+  const client = getClient()
+  const dispatch = createEventDispatcher()
+  const hierarchy = client.getHierarchy()
+  let label = ''
+  let space: Ref<Project>
+
+  $: _class = currentSpace ? hierarchy.getClass(currentSpace._class).label : undefined
+  let classLabel = ''
+  $: {
+    const doc = docs[0]
+    if (space === undefined) space = doc.space
+    translate(hierarchy.getClass(doc._class).label, {}).then((res) => (label = res.toLocaleLowerCase()))
+  }
+  $: _class && translate(_class, {}).then((res) => (classLabel = res.toLocaleLowerCase()))
+
+  async function move (doc: Issue): Promise<void> {
+    const spaceObject = await client.findOne(tracker.class.Project, { _id: space })
+    if (spaceObject === undefined) {
+      throw new Error('Move: state not found')
+    }
+    const lastOne = await client.findOne(tracker.class.Issue, {}, { sort: { rank: SortingOrder.Descending } })
+    const incResult = await client.updateDoc(
+      tracker.class.Project,
+      core.space.Space,
+      space,
+      {
+        $inc: { sequence: 1 }
+      },
+      true
+    )
+    await moveIssueToSpace(client, doc, space, {
+      status: spaceObject.defaultIssueStatus,
+      rank: calcRank(lastOne, undefined),
+      number: (incResult as any).object.sequence
+    })
+
+    dispatch('close')
+  }
+
+  const moveAll = async () => {
+    docs.forEach(async (doc) => await move(doc))
+  }
+
+  async function getSpace (): Promise<void> {
+    client.findOne(tracker.class.Project, { _id: space }).then((res) => (currentSpace = res))
+  }
+
+  async function invokeValidate (
```

Comment:
We don't have object validator for issue, and I think won't have in observable future